src/overlays/actors/ovl_Obj_Grass/z_obj_grass.c

```diff
@@ -194,82 +196,80 @@ void func_809A983C(ObjGrass* this, PlayState* play) {
-        D_809AADC0[i] = NULL;
-        D_809AAE10[i] = 422500.0f;
+    for (i = 0; i < ARRAY_COUNT(sNearestGrassElements); i++) {
+        sNearestGrassElements[i] = NULL;
+        sNearestGrassElementsDist[i] = SQ(650.0f);
     }
 
-    for (i = 0; i < ARRAY_COUNT(D_809AADA0); i++) {
-        D_809AADA0[i] = NULL;
-        D_809AADB0[i] = 422500.0f;
+    for (i = 0; i < ARRAY_COUNT(sNearestGrassGroups); i++) {
+        sNearestGrassGroups[i] = NULL;
+        sNearestGrassGroupsDist[i] = SQ(650.0f);
     }
 
-    for (i = 0; i < this->unk_2944; i++) {
-        ptr = &this->unk_144[i];
+    for (i = 0; i < this->activeGrassGroups; i++) {
+        grassGroup = &this->grassGroups[i];
 
-        for (j = 0; j < ptr->unk_FC; j++) {
-            ptr->unk_0C[j].unk_0F &= (u16)~2;
+        for (j = 0; j < grassGroup->count; j++) {
+            grassGroup->elements[j].elem_flags &= (u16)~2;
         }
     }
 
-    for (i = 0; i < this->unk_2944; i++) {
-        ptr = &this->unk_144[i];
-        temp_f0 = Math3D_Vec3fDistSq(&ptr->unk_00, &player->actor.world.pos);
+    for (i = 0; i < this->activeGrassGroups; i++) {
+        grassGroup = &this->grassGroups[i];
+        dist = Math3D_Vec3fDistSq(&grassGroup->homePos, &player->actor.world.pos);
 
-        for (j = 0; j < ARRAY_COUNT(D_809AADB0); j++) {
-            if (temp_f0 < D_809AADB0[j]) {
+        for (j = 0; j < ARRAY_COUNT(sNearestGrassGroupsDist); j++) {
+            if (dist < sNearestGrassGroupsDist[j]) {
                 break;
             }
         }
 
-        if (j < ARRAY_COUNT(D_809AADB0)) {
-            for (x = 2; x >= j; x--) {
-                D_809AADB0[x + 1] = D_809AADB0[x];
-                D_809AADA0[x + 1] = D_809AADA0[x];
+        if (j < ARRAY_COUNT(sNearestGrassGroupsDist)) {
+            for (x = ARRAY_COUNT(sNearestGrassGroupsDist) - 2; x >= j; x--) {
+                sNearestGrassGroupsDist[x + 1] = sNearestGrassGroupsDist[x];
+                sNearestGrassGroups[x + 1] = sNearestGrassGroups[x];
             }
 
-            D_809AADB0[j] = temp_f0;
-            D_809AADA0[j] = ptr;
+            sNearestGrassGroupsDist[j] = dist;
+            sNearestGrassGroups[j] = grassGroup;
         }
     }
 
-    this->unk_3294 = NULL;
+    this->carryGrassElem = NULL;
 
-    for (i = 0; i < ARRAY_COUNT(D_809AADA0); i++) {
-        ptr = D_809AADA0[i];
+    for (i = 0; i < ARRAY_COUNT(sNearestGrassGroups); i++) {
+        grassGroup = sNearestGrassGroups[i];
 
-        if (ptr != NULL) {
-            for (j = 0; j < ptr->unk_FC; j++) {
-                if (!(ptr->unk_0C[j].unk_0F & 4)) {
-                    temp_f0 = Math3D_Vec3fDistSq(&ptr->unk_0C[j].unk_00, &player->actor.world.pos);
+        if (grassGroup != NULL) {
+            for (j = 0; j < grassGroup->count; j++) {
+                if (!(grassGroup->elements[j].elem_flags & 4)) {
+                    dist = Math3D_Vec3fDistSq(&grassGroup->elements[j].pos, &player->actor.world.pos);
```

Comment:
```suggestion
                    distSq = Math3D_Vec3fDistSq(&grassGroup->elements[j].pos, &player->actor.world.pos);
```
Since this is squared, there's a few names that could reflect that:
`sNearestGrassElementsDist` -> `sNearestGrassElementsDistSq`
`2500.0f` -> `SQ(50.0f)`